cpp/tests/io/json_test.cpp

```diff
@@ -642,7 +648,9 @@ TEST_P(JsonReaderParamTest, JsonLinesDtypeInference)
   std::string data          = is_row_orient_test(test_opt) ? row_orient : record_orient;
 
   cudf::io::json_reader_options in_options =
-    cudf::io::json_reader_options::builder(cudf::io::source_info{data.data(), data.size()})
+    cudf::io::json_reader_options::builder(
+      cudf::io::source_info{cudf::host_span<std::byte const>(
+        reinterpret_cast<std::byte const*>(data.data()), data.size())})
```

Comment:
would this work?
```suggestion
      cudf::io::source_info{cudf::host_span(data.data(), data.size())})
```

Reply:
No, I tried this and variations multiple times and it was unable to determine what to do.